### DISCUSSION TITLE
Bluetooth A2DP

### DISCUSSION THREAD

**Discussion by HobmanX** [pschatzmann/arduino-audio-tools#1561] [Q&A] [answered]
I was trying to get some of the examples for Bluetooth and A2DP working, but when compiling no matter what example it keeps asking for portAudio.h and other portAudio library files. I downloaded the zip for portaudio and put it in sketchbook libraries but it still cannot find it..... no such file or directory.    Has things changed, I thought I had at least some of these examples working some time ago, but cannot get them working now..?

**Comment by pschatzmann** [comment, marked as answer]
You are somehow completely off the track!

You would use an output to PortAudio only if you compile and run a sketch via cmake on the Desktop: But A2DP is only working on an ESP32 and is definitely not supported on the Desktop.

So, what the hell are you doing ?

**Comment by HobmanX** [comment]
Looks like I have upset the "Apple Cart" or "Spilled the Beans along with alot of Brown Gravy or Tomato Sauce"...??
I had a few old sketches as I thought bluetooth sketches in a folder that I thought were working before but yesterday I couldnt get them to compile giving me an error about PortAudio.  I never have tried to use the Desktop options...so don't know.
I discovered I had a ESP32A2DP library in my libraries folder with some examples......these compiled OK...but for my ESP32-PICO-D4 as mostly too big to compile and run..??
I have asked many times before.... Is there anywhere I can get info to read on how this all works.. I cannot make much sense out of the info on this forum or Wiki it just describes individual bits and examples but does not describe how it is all put together...?
